fuel-reactor/src/main/kotlin/com/github/kittinunf/fuel/reactor/Reactor.kt

```diff
@@ -0,0 +1,51 @@
+package com.github.kittinunf.fuel.reactor
+
+import com.github.kittinunf.fuel.core.Deserializable
+import com.github.kittinunf.fuel.core.FuelError
+import com.github.kittinunf.fuel.core.Request
+import com.github.kittinunf.fuel.core.Response
+import com.github.kittinunf.fuel.core.deserializers.ByteArrayDeserializer
+import com.github.kittinunf.fuel.core.deserializers.StringDeserializer
+import com.github.kittinunf.fuel.core.response
+import com.github.kittinunf.result.Result
+import reactor.core.publisher.Mono
+import java.nio.charset.Charset
+
+private fun <T : Any> Request.monoOfResultFold(mapper: Deserializable<T>): Mono<T> =
+    Mono.create<T> { sink ->
+        sink.onCancel(this::cancel)
+        val (_, _, result) = response(mapper)
+        result.fold(sink::success, sink::error)
+    }
+
+fun Request.monoOfBytes(): Mono<ByteArray> =
```

Comment:
[Q] Shall we omit the `Of` here so it kinda goes similar to the `responseXXX` version? 

`responseBytes` => `monoBytes`
`responseString` => `monoString`
`responseObject` => `monoObject`

Reply:
I prefer `monoOf`, but I agree that using `mono` only can be better for consistency. Changed in f669d3c.

Reply:
`monoOf<T>` would still work, and agreed with all the other comments 🗡